src/ansi-c/parser.y

```diff
@@ -3407,6 +3407,19 @@ cprover_function_contract_sequence_opt:
           /* nothing */
           { init($$); }
         | cprover_function_contract_sequence
+        {
+          // Function contracts should either be attached to a
+          // top-level function declaration or top-level function
+          // definition.  Any embedded function pointer scopes should
+          // be disallowed.
+          int contract_in_global_scope = (PARSER.scopes.size() == 1);
+          int contract_in_top_level_function_scope = (PARSER.scopes.size() == 2);
```

Comment:
Could you explain the magic numbers `1` and `2` in the comment?

Reply:
I was hoping the long variable names, combined with the comment above, would be self-documenting.  When we initialize the parser, we start by pushing a global scope onto the stack.  (There's an invariant that we never have an empty scopes stack.)  So a size of 1 means the stack only contains the global scope.  A size of two means we must be in some top-level event like a function declaration or definition because we've pushed something else on the stack.